transports/http/src/main/java/org/mule/transport/http/HttpMuleMessageFactory.java

```diff
@@ -165,8 +168,16 @@ else if (transportMessage instanceof HttpMethod)
         httpHeaders.put(HttpConnector.HTTP_HEADERS, new HashMap<String, Object>(headers));
 
         String encoding = getEncoding(headers);
-        
-        queryParameters.put(HttpConnector.HTTP_QUERY_PARAMS, processQueryParams(uri, encoding));
+
+        try
+        {
+            queryParameters.put(HttpConnector.HTTP_QUERY_PARAMS, processQueryParams(uri, encoding));
+        }
+        catch (IllegalArgumentException e)
+        {
+            Message errorDescription = MessageFactory.createStaticMessage(e.getMessage());
```

Comment:
Shouldn't you use a custom message saying the query parameters were invalid? The exception message will still be available.